scripts/tools/nrfconnect/generate_nrfconnect_chip_factory_data.py

```diff
@@ -244,8 +246,8 @@ def _validate_args(self):
             "Cannot find Spake2+ verifier, to generate a new one please provide passcode (--passcode)"
         assert ((self._args.gen_certs and self._args.chip_cert_path) or (self._args.dac_cert and self._args.pai_cert and self._args.dac_key)), \
             "Cannot find paths to DAC or PAI certificates .der files. To generate a new ones please provide a path to chip-cert executable (--chip_cert_path) and add --gen_certs argument"
-        assert self._args.output.endswith(".json"), \
-            "Output path doesn't contain .json file path. ({})".format(self._args.output)
+        assert not (self._args.output.endswith(".json")), \
+            "Output path contain .json extension in path that is redundant. ({})".format(self._args.output)
```

Comment:
If we want to call assert on that, this should point out that the .json extension in the path is *forbidden* instead of *redundant*. 
Apart from that, I think that acutally it is redundant, but the script should remove the extension to keep backward compatibility.

Reply:
It is mentioned that the .json should not be provided in the argument description:

The .json extension will be automatically added by the script and should not be provided.

Do you mean we should support both versions - with and without extension?

Reply:
Yes, I meant to support both ways because in the other case, we break compatibility with the previous versions of the scripts. I guess it is a one-liner but it will help customers that may want to use this script as previously.

Reply:
Ok